Jenkinsfile_nightly

```diff
@@ -0,0 +1,203 @@
+'''
+    }
+
+    after('test'){
+        junit 'application/build/test-results/test/**/*.xml'
+        publishHTML([
+            allowMissing         : true,
+            alwaysLinkToLastBuild: true,
+            keepAll              : true,
+            reportDir            : "application/build/reports/tests/test/",
+            reportFiles          : 'index.html',
+            reportName           : 'Unit Test Report'
+        ])
+
+        publishHTML([
+            allowMissing         : true,
+            alwaysLinkToLastBuild: true,
+            keepAll              : true,
+            reportDir            : "application/build/reports/checkstyle/",
+            reportFiles          : 'main.html',
+            reportName           : 'Checkstyle Main Report'
+        ])
+        publishHTML([
+            allowMissing         : true,
+            alwaysLinkToLastBuild: true,
+            keepAll              : true,
+            reportDir            : "application/build/reports/checkstyle/",
+            reportFiles          : 'test.html',
+            reportName           : 'Checkstyle Test Report'
+        ])
+        publishHTML([
+            allowMissing         : true,
+            alwaysLinkToLastBuild: true,
+            keepAll              : true,
+            reportDir            : "application/build/reports/pmd/",
+            reportFiles          : 'main.html',
+            reportName           : 'PMD Main Report'
+        ])
+        publishHTML([
+            allowMissing         : true,
+            alwaysLinkToLastBuild: true,
+            keepAll              : true,
+            reportDir            : "application/build/reports/pmd/",
+            reportFiles          : 'test.html',
+            reportName           : 'PMD Test Report'
+        ])
+
+        try {
+            sh './gradlew jacocoTestReport --info'
+        }finally {
+            jacoco(execPattern: 'application/build/jacoco/test.exec', buildOverBuild: false,
+                exclusionPattern: '**/test/*, ' +
+                    '**/uk/gov/hmcts/dm/DmApp.java,'+
+                    '**/uk/gov/hmcts/dm/hateos/*,'+
+                    '**/uk/gov/hmcts/dm/exception/*,'+
+                    '**/uk/gov/hmcts/dm/domain/*,'+
+                    '**/uk/gov/hmcts/dm/commandobject/*,'+
+                    '**/uk/gov/hmcts/dm/hibernate/*,'+
+                    '**/uk/gov/hmcts/dm/config/**/*,'+
+                    '**/uk/gov/hmcts/dm/errorhandler/*,'+
+                    '**/uk/gov/hmcts/dm/repository/RepositoryFinder.java')
+            publishHTML([
+                allowMissing         : true,
+                alwaysLinkToLastBuild: true,
+                keepAll              : true,
+                reportDir            : "application/build/reports/jacoco/test/html/",
+                reportFiles          : 'index.html',
+                reportName           : 'Jacoco Coverage Report'
+            ])
+        }
+    }
+
+    after('smoketest:demo') {
```

Comment:
Everything starting from this line should not be needed for nightly builds as there's no deployment, smoke or functional tests